oci/oci_test.go

```diff
@@ -0,0 +1,71 @@
+package oci
+
+import (
+	"fmt"
+	"github.com/stretchr/testify/assert"
+	"testing"
+)
+
+type testDataSplitReference struct {
+	ref   string
+	dir   string
+	image *string
+}
+
+type testDataScopeValidation struct {
+	scope      string
+	dir        string
+	errMessage string
+}
+
+func TestSplitReferenceIntoDirAndImageWindows(t *testing.T) {
+	image1 := "busybox:latest"
+	image2 := "busybox"
+	image3 := ""
+	tests := []testDataSplitReference{
+		{`C:\foo\bar:busybox:latest`, `C:\foo\bar`, &image1},
+		{`C:\foo\bar:busybox`, `C:\foo\bar`, &image2},
+		{`C:\foo\bar:`, `C:\foo\bar`, &image3},
+		{`C:\foo\bar`, `C:\foo\bar`, nil},
+	}
+	for _, test := range tests {
+		dir, image := splitDirAndImageWindows(test.ref)
+		assert.Equal(t, test.dir, dir, "Unexpected OCI directory")
+		assert.Equal(t, test.image, image, "Unexpected image")
+	}
+}
+
+func TestSplitReferenceIntoDirAndImageNonWindows(t *testing.T) {
+	image1 := "busybox:latest"
+	image2 := "busybox"
+	image3 := ""
+	tests := []testDataSplitReference{
+		{"/foo/bar:busybox:latest", "/foo/bar", &image1},
+		{"/foo/bar:busybox", "/foo/bar", &image2},
+		{"/foo/bar:", "/foo/bar", &image3},
+		{"/foo/bar", "/foo/bar", nil},
+	}
+	for _, test := range tests {
+		dir, image := splitDirAndImageNonWindows(test.ref)
+		assert.Equal(t, test.dir, dir, "Unexpected OCI directory")
+		assert.Equal(t, test.image, image, "Unexpected image")
+	}
+}
+
+func TestValidateScopeWindows(t *testing.T) {
+	tests := []testDataScopeValidation{
+		{`C:\foo:busybox`, `C:\foo`, ""},
+		{`D:\:apline`, `D:\`, ""},
+		{"C:", "C:", "Invalid scope 'C:'. Must be an absolute path"},
+		{"E", "E", "Invalid scope 'E'. Must be an absolute path"},
+		{"", "", "Invalid scope ''. Must be an absolute path"},
```

Comment:
(`ValidatePolicyConfigurationScope` is never called with the `""` value, so this case is unnecessary. Doesn’t hurt though.)

Reply:
ok